src/managed/Microsoft.Extensions.DependencyModel/DependencyContextLoader.cs

```diff
@@ -126,6 +126,41 @@ private DependencyContext LoadAssemblyContext(Assembly assembly, IDependencyCont
 
             return null;
         }
+
+        private string GetDepsJsonPath(Assembly assembly)
+        {
+            string depsJsonFile = Path.ChangeExtension(assembly.Location, DepsJsonExtension);
+            bool depsJsonFileExists = _fileSystem.File.Exists(depsJsonFile);
+
+            if (!depsJsonFileExists)
+            {
+                // in some cases (like .NET Framework shadow copy) the Assembly Location 
+                // and CodeBase will be different, so also try the CodeBase
+                string assemblyCodeBase = GetNormalizedCodeBasePath(assembly);
+                if (!string.IsNullOrEmpty(assemblyCodeBase) &&
+                    assembly.Location != assemblyCodeBase)
+                {
+                    depsJsonFile = Path.ChangeExtension(assemblyCodeBase, DepsJsonExtension);
+                    depsJsonFileExists = _fileSystem.File.Exists(depsJsonFile);
+                }
+            }
+
+            return depsJsonFileExists ?
+                depsJsonFile :
+                null;
+        }
+
+        private static string GetNormalizedCodeBasePath(Assembly assembly)
+        {
+            try
+            {
+                return new Uri(assembly.CodeBase).LocalPath;
```

Comment:
You could also do `if (Uri.TryCreate(assembly.CodeBase, UriKind.Absolute, out var uri) && uri.IsLocal) { return uri.LocalPath; }` to avoid the `try-catch`

Reply:
Done.